pandas/tests/indexes/period/test_period.py

```diff
@@ -649,6 +649,72 @@ def test_insert(self):
             result = period_range("2017Q1", periods=4, freq="Q").insert(1, na)
             tm.assert_index_equal(result, expected)
 
+    def test_contains_raise_error_if_period_index_is_in_multi_index(self):
+        # issue 20684
+        df = DataFrame(
+            {
+                "A": [Period("2019"), "x1"],
+                "B": ["y1", Period("2018")],
+                "C": ["z1", "z2"],
+                "V1": [1, 2],
+                "V2": [10, 20],
+            }
+        ).set_index(["A", "B", "C"])
+        msg = r"Period\('2019', 'A-DEC'\), 'foo', 'bar'"
+        with pytest.raises(KeyError, match=msg):
+            df.loc[(Period("2019"), "foo", "bar")]
+
+        msg = r"'foo', Period\('2018', 'A-DEC'\), 'bar'"
+        with pytest.raises(KeyError, match=msg):
+            df.loc[("foo", Period("2018"), "bar")]
+
+        pbc = DataFrame(
+            {"p": [Period("2017")], "b": ["b1"], "c": ["c1"], "v": [1]}
+        ).set_index(["p", "b", "c"])
+        msg = r"Period\('2017', 'A-DEC'\), 'wibble', 'c1'"
+        with pytest.raises(KeyError, match=msg):
+            pbc.loc[(Period("2017"), "wibble", "c1")]
+
+        msg = r"Period\('2017', 'A-DEC'\), 'b1', 'wobble'"
+        with pytest.raises(KeyError, match=msg):
+            pbc.loc[(Period("2017"), "b1", "wobble"), "v"]
+
+        ybc = DataFrame({"y": ["2017"], "b": ["b1"], "c": ["c1"], "v": [1]}).set_index(
+            ["y", "b", "c"]
+        )
+        assert ybc.loc[("2017", "b1", "c1"), "v"] == 1
+
+        msg = r"'2017', 'b1', 'wobble'"
+        with pytest.raises(KeyError, match=msg):
+            ybc.loc[("2017", "b1", "wobble"), "v"]
+
+        pb = DataFrame({"p": [Period("2017")], "b": ["b1"], "v": [1]}).set_index(
+            ["p", "b"]
+        )
+        assert pb.loc[(Period("2017"), "b1"), "v"] == 1
```

Comment:
can you add some comments on groups of tests here. really hard to tell why things *should* fail as there are a lot of tests. is there any way to paramterize these? these seems duplicated (but hard to tell)